File: php/features.php
<?php
declare(strict_types=1);

// CoffeeTwo SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class CoffeeTwoFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new CoffeeTwoBaseFeature();
            case "test":
                return new CoffeeTwoTestFeature();
            default:
                return new CoffeeTwoBaseFeature();
        }
    }
}

File: py/features.py
# CoffeeTwo SDK feature factory

from feature.base_feature import CoffeeTwoBaseFeature
from feature.test_feature import CoffeeTwoTestFeature


def _make_feature(name):
    features = {
        "base": lambda: CoffeeTwoBaseFeature(),
        "test": lambda: CoffeeTwoTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
